lib/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/viewmodels/delegates_bloc.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:seeds/datasource/remote/model/delegate_model.dart';
+import 'package:seeds/datasource/remote/model/member_model.dart';
+import 'package:seeds/domain-shared/page_command.dart';
+import 'package:seeds/domain-shared/page_state.dart';
+import 'package:seeds/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/mapper/delegates_load_data_mapper.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/mapper/remove_delegate_mapper.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/usecase/delegates_load_data_usecase.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/usecase/load_delegate_as_member_use_case.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/usecase/remove_delegate_use_case.dart';
+import 'package:seeds/screens/explore_screens/vote_screens/delegate/delegates_tab/interactor/viewmodels/delegates_page_commands.dart';
+
+part 'delegates_event.dart';
+part 'delegates_state.dart';
+
+class DelegatesBloc extends Bloc<DelegatesEvent, DelegatesState> {
+  DelegatesBloc() : super(DelegatesState.initial()) {
+    on<LoadDelegatesData>(_loadDelegatesData);
+    on<InitOnboardingDelegate>(_initOnboardingDelegate);
+    on<OnRemoveDelegateTapped>(_onRemoveDelegateTapped);
+    on<RemoveDelegate>(_removeDelegate);
+    on<ClearDelegatesPageCommand>((_, emit) => emit(state.copyWith()));
+  }
+
+  Future<void> _loadDelegatesData(LoadDelegatesData event, Emitter<DelegatesState> emit) async {
+    emit(state.copyWith(pageState: PageState.loading));
+    final Result result = await DelegatesLoadDataUseCase().run();
+    if (!result.isError && result.asValue != null && result.asValue!.value is DelegateModel) {
+      final DelegateModel delegate = result.asValue!.value;
+      if (delegate.delegatee.isNotEmpty) {
+        final Result memberResult = await LoadDelegateAsMemberUseCase().run(delegateAccount: delegate.delegatee);
+        emit(DelegatesLoadDataStateMapper().mapResultToState(state, memberResult));
```

Comment:
emit vs yield?

Reply:
I think is for the sake of simplicity
> Why Bloc v7.2.0?
> 
> We had been wrestling with an issue in Dart for a while that impacted the predictability of state changes when using bloc. We were waiting to see if this would be addressed in the Dart SDK, but because it would require changes to asynchronous constructs including yield* (yield-each), it would be risky and unlikely to be addressed in the near future. Because predictability is at the core of the bloc library, we started to experiment with alternative approaches.
> 
> We knew that in order to address the predictability issue, we had to move away from async generators and introduce an API which gave us granular control over when state changes occur. In the current version, mapEventToState returns a Stream which lends itself to relying on async generators (async*) in order to react to events and emit states. The new on<Event> API introduced in v7.2.0 makes it easier and safer to implement complex state changes while eliminating uncertainty.